stdlib/public/core/Reverse.swift

```diff
@@ -12,34 +12,47 @@
 
 /// An index that traverses the same positions as an underlying index,
 /// with inverted traversal direction.
-public struct ReverseIndex<Base> : Comparable {
-  public init(_ base: Base) { self.base = base }
```

Comment:
I know this is a public API so changing it could be problematic but we are making a lot of breaking changes... how should I think about API contracts when making changes?